packages/twenty-server/src/database/commands/upgrade-version-command/1-2/1-2-assign-roles-to-existing-api-keys.command.ts

```
@@ -0,0 +1,164 @@
+    }
+
+    this.logger.log(`    Found ${apiKeys.length} API keys in workspace`);
+
+    const apiKeyIds = apiKeys.map((key) => key.id);
+    const existingRoleTargets = await this.roleTargetsRepository.find({
+      where: {
+        workspaceId,
+        apiKeyId: In(apiKeyIds),
+      },
+    });
+
+    const apiKeysWithRoles = new Set(
+      existingRoleTargets.map((rt) => rt.apiKeyId),
+    );
+    const apiKeysWithoutRoles = apiKeys.filter(
+      (key) => !apiKeysWithRoles.has(key.id),
+    );
+
+    if (apiKeysWithoutRoles.length === 0) {
+      this.logger.log(`    All API keys already have role assignments`);
+
+      return { processed: apiKeys.length, assigned: 0 };
+    }
+
+    this.logger.log(
+      `    Found ${apiKeysWithoutRoles.length} API keys without role assignments`,
+    );
+
+    if (dryRun) {
+      this.logger.log(
+        `    [DRY RUN] Would assign Admin roles to ${apiKeysWithoutRoles.length} API keys`,
+      );
+
+      return {
+        processed: apiKeys.length,
+        assigned: apiKeysWithoutRoles.length,
+      };
+    }
+
+    const adminRole = await this.roleRepository.findOne({
+      where: {
+        workspaceId,
+        label: ADMIN_ROLE_LABEL,
+      },
+    });
+
+    if (!adminRole) {
+      this.logger.warn(
+        `    No Admin role found in workspace ${workspaceId}, skipping...`,
+      );
+
+      return { processed: apiKeys.length, assigned: 0 };
+    }
+
+    this.logger.log(`    Using Admin role: ${adminRole.id}`);
+
+    let assignedCount = 0;
+
+    for (const apiKey of apiKeysWithoutRoles) {
+      try {
+        await this.apiKeyRoleService.assignRoleToApiKey({
+          apiKeyId: apiKey.id,
+          roleId: adminRole.id,
+          workspaceId,
+        });
+
+        this.logger.log(
+          `      ✅ Assigned Admin role to API key "${apiKey.name}" (${apiKey.id})`,
+        );
+        assignedCount++;
+      } catch (error) {
+        this.logger.error(
```

error will be swallowed here, while you seem to wanted it to throw above

```
if (!options.dryRun) {
        throw error;
      }
```

Not sure what is the best option though, i don't think we want the whole script to fail each time we fail to migrate an api key, but maybe rather to keep a list of failed workspaces ids and log it at the end of the command for us to debug. wdyt?